tracing-subscriber/src/fmt/format/json.rs

```diff
@@ -236,6 +236,28 @@ where
                 serializer.serialize_entry("target", meta.target())?;
             }
 
+            if self.display_thread_name {
+                match std::thread::current().name() {
+                    Some(name) => {
+                        serializer.serialize_entry("threadName", name)?;
+                    }
+                    None => {
+                        // fall-back to thread id when name is absent and id's are not enabled
+                        if !self.display_thread_id {
+                            serializer.serialize_entry(
+                                "threadId",
+                                &format!("{:?}", std::thread::current().id()),
```

Comment:
Using `format!` here allocates, since `format!` constructs a new `String`. It would be nice if that was not necessary. 

After looking at the `serde` API docs, though, it seems like it might be --- there's no way to serialize something's `Debug` implementation as a string value except by turning it into a `String` and borrowing it. That seems like a deficiency in `serde` (or, I'm missing something in the  `serde` API).

Reply:
Yeah, I did try searching a way for serde to directly use `Debug` impl but could not find how. So, had to use `format!`